packages/regenesis-surgery/scripts/contract-recompiler.ts

```diff
@@ -0,0 +1,174 @@
+import solc from 'solc'
+import linker from 'solc/linker'
+import { createReadStream } from 'fs'
+import { parseChunked } from '@discoveryjs/json-ext'
+import dotenv from 'dotenv'
+import path from 'path'
+
+dotenv.config()
+
+const env = process.env
+const STATE_DUMP_PATH = env.STATE_DUMP_PATH
+const ETHERSCAN_CONTRACTS_PATH = env.ETHERSCAN_CONTRACTS_PATH
+
+interface EtherscanContract {
+  contractAddress: string
+  code: string
+  hash: string
+  sourceCode: string
+  creationCode: string
+  contractFileName: string
+  contractName: string
+  compilerVersion: string
+  optimizationUsed: string
+  runs: string
+  constructorArguments: string
+  library: string
+}
+
+;(async () => {
+  const etherscanContracts: EtherscanContract[] = await parseChunked(
+    createReadStream(ETHERSCAN_CONTRACTS_PATH)
+  )
+
+  // Corresponds to vanilla solidity
+  const compilerVersionsToSolc = {
+    'v0.5.16': 'v0.5.16+commit.9c3226ce',
+    'v0.5.16-alpha.7': 'v0.5.16+commit.9c3226ce',
+    'v0.6.12': 'v0.6.12+commit.27d51765',
+    'v0.7.6': 'v0.7.6+commit.7338295f',
+    'v0.7.6+commit.3b061308': 'v0.7.6+commit.7338295f', // what vanilla solidity should this be?
+    'v0.7.6-allow_kall': 'v0.7.6+commit.7338295f', // ^same q
+    'v0.8.4': 'v0.8.4+commit.c7e474f2',
```

Comment:
@ben-chain can i just use the default solc binary for all the different ovm versions? or should they correspond to slightly different ones?